runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -170,106 +191,35 @@ public class StreamingDataflowWorker {
-  static final int NUM_COMMIT_STREAMS = 1;
-  static final int GET_WORK_STREAM_TIMEOUT_MINUTES = 3;
-  static final Duration COMMIT_STREAM_TIMEOUT = Duration.standardMinutes(1);
-
   private static final int DEFAULT_STATUS_PORT = 8081;
-
   // Maximum size of the result of a GetWork request.
   private static final long MAX_GET_WORK_FETCH_BYTES = 64L << 20; // 64m
-
   // Reserved ID for counter updates.
   // Matches kWindmillCounterUpdate in workflow_worker_service_multi_hubs.cc.
   private static final String WINDMILL_COUNTER_UPDATE_WORK_ID = "3";
-
   /** Maximum number of failure stacktraces to report in each update sent to backend. */
   private static final int MAX_FAILURES_TO_REPORT_IN_UPDATE = 1000;
 
-  // TODO(https://github.com/apache/beam/issues/19632): Update throttling counters to use generic
-  // throttling-msecs metric.
-  public static final MetricName BIGQUERY_STREAMING_INSERT_THROTTLE_TIME =
-      MetricName.named(
-          "org.apache.beam.sdk.io.gcp.bigquery.BigQueryServicesImpl$DatasetServiceImpl",
-          "throttling-msecs");
-
   private static final Duration MAX_LOCAL_PROCESSING_RETRY_DURATION = Duration.standardMinutes(5);
-
-  /** Returns whether an exception was caused by a {@link OutOfMemoryError}. */
-  private static boolean isOutOfMemoryError(Throwable t) {
-    while (t != null) {
-      if (t instanceof OutOfMemoryError) {
-        return true;
-      }
-      t = t.getCause();
-    }
-    return false;
-  }
-
-  private static MapTask parseMapTask(String input) throws IOException {
-    return Transport.getJsonFactory().fromString(input, MapTask.class);
-  }
-
-  public static void main(String[] args) throws Exception {
-    JvmInitializers.runOnStartup();
-
-    DataflowWorkerHarnessHelper.initializeLogging(StreamingDataflowWorker.class);
-    DataflowWorkerHarnessOptions options =
-        DataflowWorkerHarnessHelper.initializeGlobalStateAndPipelineOptions(
-            StreamingDataflowWorker.class);
-    DataflowWorkerHarnessHelper.configureLogging(options);
-    checkArgument(
-        options.isStreaming(),
-        "%s instantiated with options indicating batch use",
-        StreamingDataflowWorker.class.getName());
-
-    checkArgument(
-        !DataflowRunner.hasExperiment(options, "beam_fn_api"),
-        "%s cannot be main() class with beam_fn_api enabled",
-        StreamingDataflowWorker.class.getSimpleName());
-
-    StreamingDataflowWorker worker =
-        StreamingDataflowWorker.fromDataflowWorkerHarnessOptions(options);
-
-    // Use the MetricsLogger container which is used by BigQueryIO to periodically log process-wide
-    // metrics.
-    MetricsEnvironment.setProcessWideContainer(new MetricsLogger(null));
-
-    JvmInitializers.runBeforeProcessing(options);
-    worker.startStatusPages();
-    worker.start();
-  }
-
+  private static final Random clientIdGenerator = new Random();
+  final WindmillStateCache stateCache;
```

Comment:
can this be private final? and stay in same location?

Reply:
this is accessed in the StreamingDataflowWorkerTest.testMergeWindowsCaching, i can mark it as `@VisibleForTesting`?

I plan to do some breakup/cleanup of StreamingDataflowWorker in a later CL.  If we opt to use dependency injection and inject the cache for tests, we can make all of this private.

Reply:
I was thinking we can eventually have either StreamingApplianceDataflowWorker or StreamingEngineDataflowWorker (depending on the StreamingDataflowWorkerOptions).  StreamingDataflowWorker can instantiate either one depending on the passed in options, and we can have different tests for each.

We can inject all the dependencies into the constructors and have some static factory class/methods to create the shapes we want (for test or not etc). 

Future code changes won't ever touch the explicit code paths/classes for appliance.

Reply:
Also for my own knowledge, why don't we use Guice or Dagger dependency injection (if we want to avoid the runtime overhead)? @scwhittle